src/main/java/wordcloud/ParallelLayeredWordCloud.java

```diff
@@ -0,0 +1,111 @@
+package wordcloud;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+
+/**
+ * A LayeredWordCloud which can process each layer in its own Thread, thus
+ * minimizing processing time.
+ * 
+ * @author &#64;wolfposd
+ */
+public class ParallelLayeredWordCloud extends LayeredWordCloud {
+
+    private static final Logger LOGGER = Logger.getLogger(ParallelLayeredWordCloud.class);
+
+    private List<Future<?>> executorFutures = new ArrayList<>();
+
+    private ExecutorService executorservice;
+
+    public ParallelLayeredWordCloud(int layers, int width, int height, CollisionMode collisionMode) {
+        super(layers, width, height, collisionMode);
+        executorservice = Executors.newFixedThreadPool(layers);
+    }
+
+    /**
+     * constructs the wordcloud specified by layer using the given
+     * wordfrequencies.<br>
+     * This is a non-blocking call.
+     * 
+     * @param layer
+     *            Wordcloud Layer
+     * @param wordFrequencies
+     *            the WordFrequencies to use as input
+     */
+    @Override
+    public void build(final int layer, final List<WordFrequency> wordFrequencies) {
+        Future<?> f = executorservice.submit(new Runnable() {

```

Comment:
this can be made final. 
I prefer to avoid single variable names with the exception of iterators.